cpp/src/join/hash_join.cu

```diff
@@ -43,51 +61,36 @@ std::pair<std::unique_ptr<table>, std::unique_ptr<table>> get_empty_joined_table
 /**
  * @brief Builds the hash table based on the given `build_table`.
  *
- * @throw cudf::logic_error if the number of columns in `build` table is 0.
- * @throw cudf::logic_error if the number of rows in `build` table is 0.
- * @throw cudf::logic_error if insertion to the hash table fails.
- *
  * @param build Table of columns used to build join hash.
+ * @param hash_table Build hash table.
  * @param compare_nulls Controls whether null join-key values should match or not.
  * @param stream CUDA stream used for device memory operations and kernel launches.
  *
- * @return Built hash table.
  */
-std::unique_ptr<multimap_type, std::function<void(multimap_type*)>> build_join_hash_table(
-  cudf::table_view const& build, null_equality compare_nulls, rmm::cuda_stream_view stream)
+void build_join_hash_table(cudf::table_view const& build,
+                           multimap_type& hash_table,
```

Comment:
Can't this return the hash_table?

Reply:
Currently infeasible due to a compiler bug (https://godbolt.org/z/7fqsK4hPr) in `cuda-11.0` and ` cuda-11.4`

Reply:
Depends on https://github.com/NVIDIA/libcudacxx/issues/205